jena-arq/src/main/java/org/apache/jena/riot/JsonLDReadContext.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.riot;
+
+import com.github.jsonldjava.core.JsonLdOptions;
+import org.apache.jena.atlas.web.ContentType;
+import org.apache.jena.riot.lang.JsonLDReader;
+import org.apache.jena.riot.system.StreamRDF;
+import org.apache.jena.sparql.util.Context;
+
+import java.io.InputStream;
+
+/**
+ * Set of parameters that can be used to control the reading of JSON-LD.
+ *
+ * This class provides setters to define a "Context" suitable to be passed as 
+ * last argument to  {@link ReaderRIOT#read(InputStream, String, ContentType, StreamRDF, Context)}
+ * when the ReaderRIOT has been created with one of the JSON-LD RDFFormat variants (that is, when it is
+ * an instance of {@link JsonLDReader})
+ *
+ * Parameters that are actually useful are ''documentLoader'' and ''produceGeneralizedRdf''.
+ *
+ */
+public class JsonLDReadContext extends Context {
```

Comment:
OK - I see the intention here but subclassing isn't going to work. This is because `Context` is also used in places where it is a container of settings from different subsystems - e.g. `RIOT.getContext()` is a global system place to have context settings. A subclass doesn't because it can't be shared.

Instead, have a library class `JsonLD` with static functions.
```
public static void setOptions(Context cxt, JsonLdOptions opts) {
        cxt.set(JsonLDReader.JSONLD_OPTIONS, opts);
}
```

Reply:
I was doing the same as it is being done in [JsonLdWriteContext](https://github.com/apache/jena/blob/master/jena-arq/src/main/java/org/apache/jena/riot/JsonLDWriteContext.java#L43). Why would it work for the writer and not for the reader? 

The only thing this class is doing is to forward the method calls to the super.context.

Reply:
OK - I see now. I hadn't remembered `JsonLdWriteContext`; I'd also jumped to the conclusion that `JsonLdReadContext` would be used where general `Context` is.

Same style as existing is fine; if is to be changed, that's a separate work item.